src/Socket/messages-recv.ts

```diff
@@ -321,6 +321,14 @@ export const makeMessagesRecvSocket = (config: SocketConfig) => {
 				msg.messageStubParameters = [ approvalMode.attrs.state ]
 			}
 
+			break
+		case 'created_membership_requests':
+			msg.messageStubType = WAMessageStubType.GROUP_MEMBERSHIP_JOIN_APPROVAL_REQUEST_NON_ADMIN_ADD
+			msg.messageStubParameters = [ 'created', child.attrs.request_method ]
+			break
+		case 'revoked_membership_requests':
+			msg.messageStubType = WAMessageStubType.GROUP_MEMBERSHIP_JOIN_APPROVAL_REQUEST_NON_ADMIN_ADD
+			msg.messageStubParameters = [ 'revoked' ]
```

Comment:
here you cannot rely on `participant` jid because this jid is from the **author** of the action, which can be different from the requesting user.

you can get the requesting user with the following:
```js
const requester = getBinaryNodeChild(child, 'participant')?.attrs.jid`
```

(you can use `getBinaryNodeChildren` if unsure about `participant` being _always_ just one node or multiple occurrences)

you can also check if it's a request retraction or denial with the following logic:
```
areJidsSameUser(requester, participant) ? 'retracted' : 'denied'
```

Reply:
100%

Reply:
applied